renormalizer/transport/kubo.py

```diff
@@ -0,0 +1,346 @@
+            self.distance_matrix = np.arange(mol_num).reshape(-1, 1) - np.arange(mol_num).reshape(1, -1)
+            self.distance_matrix[0][-1] = 1
+            self.distance_matrix[-1][0] = -1
+
+        # current caused by pure eletronic coupling
+        holstein_current_model = {}
+        # current related to phonons
+        peierls_current_model = {}
+        # loop through the Hamiltonian to construct current operator
+        for dof_names, terms in self.mol_list.model.items():
+            # find out terms that contains two electron operators
+            # idx of the dof for the model
+            dof_idx1 = dof_idx2 = None
+            # idx of the dof in `terms`
+            term_idx1 = term_idx2 = None
+            for term_idx, dof_name in enumerate(dof_names):
+                e_or_ph, dof_idx = dof_name.split("_")
+                dof_idx = int(dof_idx)
+                if e_or_ph == "e":
+                    if dof_idx1 is None:
+                        dof_idx1 = dof_idx
+                        term_idx1 = term_idx
+                    elif dof_idx2 is None:
+                        dof_idx2 = dof_idx
+                        term_idx2 = term_idx
+                    else:
+                        raise ValueError(f"The model contains three-electron (or more complex) operator for {dof_names}")
+                del term_idx, dof_name
+            # two electron operators not found. Not relevant to the current operator
+            if dof_idx1 is None or dof_idx2 is None:
+                continue
+            # two electron operators found. Relevant to the current operator
+            # at most 3 dofs are involved. More complex cases are probably supported but not tested
+            if len(dof_names) not in (2, 3):
+                raise NotImplementedError("Complex vibration potential not implemented")
+            # Holstein terms
+            if len(dof_names) == 2:
+                current_model = holstein_current_model
+            # Peierls terms
+            else:
+                current_model = peierls_current_model
+            current_model[dof_names] = []
+            # translate every term in the Hamiltonian into terms in the current operator
+            for term in terms:
+                if len(dof_names) == 3:
+                    # total term idx should be 0 + 1 + 2 = 3
+                    phonon_term_idx = 3 - term_idx1 - term_idx2
+                    assert term[phonon_term_idx].symbol in (r"b^\dagger + b", "x")
+                symbol1, symbol2 = term[term_idx1].symbol, term[term_idx2].symbol
+                if not {symbol1, symbol2} == {r"a^\dagger", "a"}:
+                    raise ValueError(f"Unknwon symbol: {symbol1}, {symbol2}")
+                if symbol1 == r"a^\dagger":
+                    factor = self.distance_matrix[dof_idx1][dof_idx2]
+                else:
+                    factor = self.distance_matrix[dof_idx2][dof_idx1]
```

Comment:
The model of BasisMultiElectron will result in an error when constructing the current operator with different input formats.
For example 
model[("e_0", )] =["a^\dagger_0 a_1", factor] 
model[("e_0", "e_1")] =["a^\dagger_0 a_1", factor] 
model[("e_0", "e_1")] =["a^\dagger", "a", factor] 
will give the same Hamiltonian, while only the last format will get a correct current operator.
I think it is ok now, because the `transport` is currently not mature. At least we should a `TODO` and `doc` here to make it clear.

Reply:
This is a good point, and I think the origin is that with the complexity of `MultiElectron`, the `key:value` structure of `model` is ill-defined. If the problem is not addressed, it will become difficult for future code to deal with the ambiguity.
When merging `MolList` and `MolList2`, I think it's a good chance to improve the interface of `MolList2`. Let's just make it compulsory to set the index inside the operator (`"a^\dagger_0"`) and use a list instead of a dict to represent the model, and `order` can be constructed from `basis` in which each every `BasisSet` knows the external dofs it represents.

----
For this PR I'll fix check the input to preclude the first example. As for the second example, a `ValueError` will be raised, which I think is well enough.

Reply:
Maybe we could add a function called `standardize_model` to make the inner format standardized.

Reply:
Yes. The two approaches can be combined. I think users will very much prefer `model=[["a^\dagger_0", "a_1", factor]]` than `model[("e_0", "e_1")] =["a^\dagger_0", "a_1", factor]` , and imagine constructing MPO by `Mpo(mol_list, [["a^\dagger_10", 1]])`. The inner standard format checking could do some sorting over list items, detect duplicate terms, etc.

Reply:
Ok.